### DISCUSSION TITLE
LocalStorage is not defined

### DISCUSSION THREAD

**Discussion by ShivaBhattacharjee** [vercel/next.js#60811] [Help]
### Summary

![Screenshot from 2024-01-18 14-58-35](https://github.com/vercel/next.js/assets/95211406/b6c65e11-0f55-4796-9395-965ea3643f7e)

Despite using the use client directive and ensuring localStorage access within a useEffect hook, I'm  encountering  LocalStorage is not defined error any suggest potential solutions?

**Comment by icyJoseph** [comment]
Hi,

`use client` is only a hint to bundlers, but it doesn't mean that the component only runs in the client side. In other words, Client Components are SSR'd.

- https://github.com/reactwg/server-components/discussions/4

That being said, if you are only accessing the `localStorage`, within a use-effect, and this happens within that scope, then you might have some privacy settings and such that are blocking access to the `localStorage`.

**Reply by ShivaBhattacharjee** [reply]
The situation is a bit complex. I've checked if it's a browser-related issue, but it isn't. Even the next build command throws the same error and fails to build.

**Reply by icyJoseph** [reply]
I see, well, 🤔, maybe it is used elsewhere? or out of your control inside another library?

**Comment by phukon** [comment]
@ShivaBhattacharjee  just use the `localStorage` object directly. `localStorage` is actually a global object in web browsers, and it doesn't necessarily require explicit access through window.

Plus it's a part of the Web Storage API and is available in the global scope, meaning it's accessible from anywhere in your JavaScript code running in a browser environment, whether it's in a function, module, or any other context. see [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage)

For an example see {GitHub https://github.com/phukon/clack/blob/main/src/context/NotesContext.tsx}
